### PR TITLE
Fix false error at BasicMessageChannel.send(null)

### PR DESCRIPTION
**Pull request by HakkyuKim**
Fixes printing misleading error messages mentioned in https://github.com/flutter-tizen/plugins/pull/78#issuecomment-829955872

Currently, calling `BasicMessageChannel.send(null)` from dart side(generally platform interface packages) causes false errors to be triggered in `ByteBufferStreamReader.ReadByte`.

This results in printing misleading error messages in the console. (You can see the error messages in `video_player` and `wakelock` examples).

```
E/ConsoleMessage(30567): Invalid read in StandardCodecByteStreamReader
E/ConsoleMessage(30567): Invalid read in StandardCodecByteStreamReader
...
```

`ByteBufferStreamReader.ReadByte` is indirectly called by `StandardMessageCodec.DecodeMessageInternal`, and in Android, the null value case is checked in a corresponding Java method `StandardMessageCodec.decodeMessage`. I'm pushing a similar patch to match the code logic.

```java
// in platform/android/io/flutter/plugin/common/StandardMessageCodec.java
@Override
public Object decodeMessage(ByteBuffer message) {
  if (message == null) {
    return null;
  }
  message.order(ByteOrder.nativeOrder());
  final Object value = readValue(message);
  if (message.hasRemaining()) {
    throw new IllegalArgumentException("Message corrupted");
  }
  return value;
}
```

As a side note, since Windows platform also exposes the `client_wrapper` to plugins, it has the same problem.

```
# on Windows cmd
Invalid read in StandardCodecByteStreamReader
Invalid read in StandardCodecByteStreamReader
...
```

So maybe this should be fixed in upstream engine? I'm not sure though.

### CODE REVIEW

File: shell/platform/common/cpp/client_wrapper/standard_codec.cc [hunk 1]
```diff
@@ -308,6 +308,9 @@ StandardMessageCodec::~StandardMessageCodec() = default;
 std::unique_ptr<EncodableValue> StandardMessageCodec::DecodeMessageInternal(
     const uint8_t* binary_message,
     size_t message_size) const {
+  if(!binary_message){
```
Comment (swift-kim) [inline shell/platform/common/cpp/client_wrapper/standard_codec.cc:311]:
```suggestion
  if (!binary_message) {
```

**Review by swift-kim — COMMENTED** [review]
> So maybe this should be fixed in upstream engine?

Yes, this change is more suitable for the upstream engine.

**Comment by HakkyuKim** [comment]
I posted an [issue](https://github.com/flutter/flutter/issues/83484) in upstream repo

**Comment by HakkyuKim** [comment]
FYI, I'm contacting Samsung's Google CLA(Contribution License Agreement) manager about making our contribution group. I'll send PR after hearing a response.

**Comment by HakkyuKim** [comment]
Our company's CLA manager is not responding to any of my emails or messages. I'll sign CLA with my personal account and look into this later. I submitted a [PR in upstream](https://github.com/flutter/engine/pull/26956) so I'll close this.

**Pull request closed without merging**